packages/inventory/src/Stocktake/DetailView/DetailView.tsx

```diff
@@ -30,16 +34,55 @@ const useDraftStocktake = () => {
     getStocktakeDetailViewApi(api)
   );
 
-  const onChangeSortBy = () => {};
+  const onChangeSortBy = (column: Column<StocktakeItem>) => {
+    dispatch(StocktakeActionCreator.sortBy(column));
+  };
 
   return { draft, save, dispatch, onChangeSortBy, sortBy: state.sortBy };
 };
 
+const Expand: FC<{ rowData: StocktakeItem }> = ({ rowData }) => {
+  return (
+    <Box p={1} height={300} style={{ overflow: 'scroll' }}>
+      <Box
+        flex={1}
+        display="flex"
+        height="100%"
+        borderRadius={4}
+        bgcolor="#c7c9d933"
+      >
+        <span style={{ whiteSpace: 'pre-wrap' }}>
+          {JSON.stringify(rowData, null, 2)}
+        </span>
+      </Box>
+    </Box>
+  );
+};
+
 export const DetailView: FC = () => {
   const { draft, save, onChangeSortBy, sortBy } = useDraftStocktake();
 
   const columns = useColumns<StocktakeItem>(
-    ['itemCode', 'itemName', GenericColumnKey.Selection],
+    [
+      'itemCode',
+      'itemName',
+      'batch',
+      'expiryDate',
+      {
+        key: 'countedNumPacks',
+        label: 'label.counted-num-of-packs',
+        width: 150,
+        align: ColumnAlign.Right,
+      },
+      {
+        key: 'snapshotNumPacks',
+        label: 'label.snapshot-num-of-packs',
+        align: ColumnAlign.Right,
+      },
```

Comment:
The list of columns is already massive. Thinking to just put reused columns there and keep single use ones defined inline 🤷

Reply:
that's a good idea